lang/java/reef-io/src/main/java/org/apache/reef/io/network/group/impl/task/OperatorTopologyStructImpl.java

```diff
@@ -234,6 +277,24 @@ private void sendToNode(final byte[] data, final ReefNetworkGroupCommProtos.Grou
         node, remove}));
     return retVal;
   }
+  
+  private byte[][] receiveArrayFromNode(final NodeStruct node, final boolean remove) {
```

Comment:
Ok if you think having the array in the name helps with readability you can keep them. But I don't think duplicating code is the way to go. Now any change that we want to do will have to be replicated everywhere there is duplication. So you need to refactor it such that there is no duplication. You can refactor the old send/receive methods to use the *Array* methods.